s3/s3.go

```diff
@@ -51,6 +51,7 @@ func initFunc(destURL string) (objectstore.ObjectStoreDriver, error) {
 		//We would depends on AWS_REGION environment variable
 		b.service.Bucket = u.Host
 	}
+	b.service.Endpoint = os.Getenv("CONVOY_S3_ENDPOINT")
```

Comment:
I think we can add an additional parameter `s3-endpoint` to `backup` command in https://github.com/rancher/convoy/blob/master/client/objectstore.go#L49 .

The new parameter can take input from the command line or environment variable like `CONVOY_S3_ENDPOINT` (which is a built-in feature with `cli` framework we're using). Then users can specify the endpoint in either way.

And we can validate the endpoint format in the input validation.

Also, since Convoy client is communicating with daemon using RPC, we need to update the RPC structure to include the new parameter as well, so it won't be applied to all the destinations even it's not needed. The requests are here https://github.com/rancher/convoy/blob/master/api/request.go#L48 .

If you want to get it tested, we can set up a new Minio server (which doesn't exist when first Convoy was written) during the testing process, and modify the test cases for S3 so we can test our S3 service easily.

Reply:
Thanks for the immediate feedback! I'll see what I can do 😄 

1. I think a new flag is great idea 👍
2. I think the CLI flag makes more sense than the environment variable actually. I'll add the flag and it's environment variable for now, but I think it's a little inconsistent if this flag is the only one with an environment variable option. Do you think both options are important?
3. I agree with endpoint format validation. Do you want input validation on the daemon side as well?
4. If I am understanding correctly, we should add a new optional `S3Endpoint string` field to all of the backup request structs. (please let me know if I've misread that 😉)
5. Sure! I've taken a quick glance at the developer guide and the integration tests. Would you like me to replace the existing S3 tests with a Minio setup or add more tests just for Minio? Any pointers you have for running the test suite would be awesome!

Reply:
@JohnStarich 

2. If you think flag only is good enough, that's totally fine.
3. Yeah, ideally both sides should validate the input.
4. Yeah, I think that's right.
5. Replace S3 test with Minio test should be enough. If minio works, no reason S3 won't work. But it's not a requirement to get your PR merged. https://github.com/rancher/convoy/blob/master/docs/development.md#integration-tests-with-aws